### PR TITLE
WIP: Make E2E tests support providerless

### PR DESCRIPTION
**Pull request by lorenz**
/kind feature

**What this PR does / why we need it**:
This makes it possible to build the E2E test suite without all of the provider-specific dependencies & code. It lowers the Bazel-built binary size from ~130MiB to ~80MiB and reduces the amount of non-K8s files included by around 1000.

**Special notes for your reviewer**:
This is marked as WIP because it may require a KEP and some additional cleanup (like dealing with the import restriction files).

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by lorenz** [comment]
CLA issues should be fixed

File: test/e2e/network/service.go [hunk 1]
```diff
@@ -1191,375 +1185,6 @@ var _ = SIGDescribe("Services", func() {
 		framework.ExpectNoError(err)
 	})
 
-	// TODO: Get rid of [DisabledForLargeClusters] tag when issue #56138 is fixed.
-	ginkgo.It("should be able to change the type and ports of a service [Slow] [DisabledForLargeClusters]", func() {
```
Comment (ZhiFeng1993) [inline test/e2e/network/service.go:1195]: Why getting rid of this test and the following one?

File: test/e2e/storage/nfs_persistent_volume-disruptive.go [hunk 2]
```diff
@@ -91,7 +91,7 @@ var _ = utils.SIGDescribe("NFSPersistentVolumes[Disruptive][Flaky]", func() {
 
 	ginkgo.BeforeEach(func() {
 		// To protect the NFS volume pod from the kubelet restart, we isolate it on its own node.
-		e2eskipper.SkipUnlessNodeCountIsAtLeast(minNodes)
+		e2eskipper.SkipUnlessNodeCountIsAtLeast(2)
```
Comment (ZhiFeng1993) [inline test/e2e/storage/nfs_persistent_volume-disruptive.go:94]: Why do we use magic number 2 here?

**Pull request closed without merging**